pyvo/dal/adhoc.py

```diff
@@ -491,14 +594,14 @@ def get_first_proc(self):
 
 class SodaRecordMixin:
     """
-    Mixin for soda functionallity for record classes.
+    Mixin for soda functionality for record classes.
     If used, it's result class must have
     `pyvo.dal.datalink.AdhocServiceResultsMixin` mixed in.
     """
     def _get_soda_resource(self):
         try:
             return self._results.get_adhocservice_by_ivoid(
-                b"ivo://ivoa.net/std/SODA#sync")
+                SODA_SYNC_IVOID.encode('utf-8'))
```

Comment:
I'd like the interface better if get_adhocservice_by_ivoid accepted strings and did the legacy encoding (where necessary) itself.

Reply:
Agree and fixed.